industrial_taxonomy/pipeline/glass_clusters/utils.py

```diff
@@ -0,0 +1,122 @@
+"""Utils to create a sector corpus"""
+from toolz import pipe
+from functools import partial
+from typing import Dict, List
+import pandas as pd
+
+from industrial_taxonomy.getters.companies_house import get_sector
+from industrial_taxonomy.getters.glass_house import (
+    get_description_tokens,
+    glass_companies_house_lookup,
+)
+
+NE_CODES = {
+    "CARDINAL",
+    "DATE",
+    "GPE",
+    "LOC",
+    "MONEY",
+    "NORP",
+    "ORDINAL",
+    "ORG",
+    "PERSON",
+    "QUANTITY",
+    "TIME",
+}
+
+
+def strip_nes(tokenised_dict: Dict[int, List[str]]) -> Dict[int, List[str]]:
+    """Removes named entities from company tokenised descriptions"""
+
+    return {
+        k: [t for t in v if all(ne not in t for ne in NE_CODES)]
+        for k, v in tokenised_dict.items()
+    }
+
+
+def reverse_dict(_dict: dict) -> Dict:
+    """Turns keys into values and viceversa"""
+    return {v: k for k, v in _dict.items()}
+
+
+def gl_sic4_lookup() -> dict:
+    """Creates a lookup between glass_ids and 4 digit SIC codes"""
+
+    return (
+        get_sector()
+        .assign(
+            glass_id=lambda df: df.index.map(
+                reverse_dict(glass_companies_house_lookup())
+            ),
+            sic4=lambda df: df["SIC5_code"].astype(str).apply(lambda x: x[:-1]),
+        )
+        .query("rank == 1")
+        .drop(axis=1, labels=["SIC5_code", "rank"])
+        .dropna(axis=0, subset=["glass_id"])
+        .set_index("glass_id")["sic4"]
+        .to_dict()
+    )
+
+
+def filter_non_matched_comps(
+    tokenised: Dict[int, List[str]], matched_ids: set
+) -> Dict[int, List[str]]:
+    """Removes tokenised descriptions of glass companies that were not matched with CH
+
+    Args:
+        tokenised: lookup between company ids and tokenised descriptions
+        matched_ids: ids from glass companies matched with CH
+
+    Returns:
+        filtered tokenised descriptions dict
+    """
+
+    return {_id: tok for _id, tok in tokenised.items() if _id in matched_ids}
+
+
+def sector_tokens_lookup(
```

Comment:
Suggest decomposing the lookup creation and the filtering into two functions - c.f. [Single Responsibility Principle](https://en.wikipedia.org/wiki/Single-responsibility_principle)

Reply:
If not decomposing in situations like this then it's important to be careful about function naming - here the function name doesn't imply it does anything about filtering.
I'd suggest at least prefixing the function name with `big_` or something to try and signal  that other things are happening.